Backpack/HorizontalNavigation/Classes/BPKHorizontalNavigationItemDefault.swift

```diff
@@ -0,0 +1,208 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018-2021 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+
+internal class BPKHorizontalNavigationItemDefault<Size: BPKHorizontalNavigationSize>: UIButton, BPKHorizontalNavigationItem {
+    struct IconDefinition {
+        let image: UIImage
+        let size: CGSize
+    }
+
+    override var isSelected: Bool {
+        didSet {
+            updateStyle()
+        }
+    }
+
+    dynamic var selectedColor: UIColor? {
+        didSet {
+            updateStyle()
+        }
+    }
+
+    var appearance: BPKHorizontalNavigationAppearance = .normal {
+        didSet {
+            updateStyle()
+        }
+    }
+
+    var icon: IconDefinition?
+
+    var hasIcon: Bool { icon != nil }
+    
+
+    /**
+     * The name to display within the item.
+     */
+    let name: String
+
+    private(set) var showNotificationDot: Bool = false
+
+    private lazy var notificationDot: UIView = {
+        let dot = UIView()
+        dot.translatesAutoresizingMaskIntoConstraints = false
+        dot.backgroundColor = .bpk_systemRed
+        dot.layer.cornerRadius = BPKSpacingMd / 2.0
+
+        return dot
+    }()
+
+    private var contentColor: UIColor {
+        if isSelected {
+            return selectedColor ?? .bpk_primary
+        } else {
+            switch (appearance) {
+            case .normal: return .bpk_textPrimary
+            case .alternate: return .bpk_skyGrayTint07
+            }
+        }
+    }
+
+    init(name: String) {
```

Comment:
I think the safest place to initialise is the initialiser, since you can have convenience initialisers in the future that define different values. I guess it's a personal preference though, no real objective argue agains it (maybe only showing the possible values on autocompletion, when you actually call this initialiser)

Reply:
I think you are right but then you have to duplicate the logic and values in two places which can cause bugs. `BPKHorizontalNavigationItemDefault` will never be exposed publicly as it's just an internal API. With the defaults duplicated there's a chance we introduce bugs by changing something only in a single place